src/EnsembleTransformKalmanInversion.jl

```diff
@@ -44,29 +48,96 @@ end
         ekp::EnsembleKalmanProcess{FT, IT, TransformInversion},
         u::AbstractMatrix{FT},
         g::AbstractMatrix{FT},
-        y::AbstractVector{FT},
-        obs_noise_cov::Union{AbstractMatrix{CT}, UniformScaling{CT}},
     ) where {FT <: Real, IT, CT <: Real}
 
 Returns the updated parameter vectors given their current values and
 the corresponding forward model evaluations.
 """
 function etki_update(
-    ekp::EnsembleKalmanProcess{FT, IT, TransformInversion{FT}},
+    ekp::EnsembleKalmanProcess{FT, IT, TransformInversion},
     u::AbstractMatrix{FT},
     g::AbstractMatrix{FT},
-    y::AbstractVector{FT},
-    obs_noise_cov::Union{AbstractMatrix{CT}, UniformScaling{CT}},
-) where {FT <: Real, IT, CT <: Real}
-    m = size(u, 2)
-    Γ_inv = ekp.process.Γ_inv
+) where {FT <: Real, IT}
+
+    y = get_obs(ekp)
+    inv_noise_scaling = ekp.Δt[end]
 
+    m = size(u, 2)
     X = FT.((u .- mean(u, dims = 2)) / sqrt(m - 1))
     Y = FT.((g .- mean(g, dims = 2)) / sqrt(m - 1))
-    Ω = inv(I + Y' * Γ_inv * Y)
-    w = FT.(Ω * Y' * Γ_inv * (y .- mean(g, dims = 2)))
+    tmp = get_buffer(get_process(ekp)) # the buffer stores Y' * Γ_inv of [size(Y,2),size(Y,1)]
```

Comment:
This whole section of code is quite difficult to read and verbose. Any way it can be simplified?

Reply:
\+ 1 . If it can't be simplified, some additional comments would also be helpful (and commented out code snippets removed)

Reply:
I think in the end it's just doing the block-wise linear algebra, and using buffers to minimize the recompute.
I think I can't "simplify" the actions but I can use some functions to make it look cleaner! I'll take a look